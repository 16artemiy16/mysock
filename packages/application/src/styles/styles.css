body {
  height: 100vh;
}

#root, .App {
  height: 100%;
  font-family: Neucha,sans-serif;
}
