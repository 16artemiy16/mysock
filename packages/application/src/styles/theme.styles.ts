type AvailableColors = 'white' | 'black' | 'red' | 'link' | 'linkActive' | 'success' | 'primary' | 'error' | 'grayLight';

export interface ThemeI {
  colors: Record<AvailableColors, string>;
}

export const theme: ThemeI = {
  colors: {
    white: 'white',
    black: 'black',
    red: 'red',

    grayLight: '#e6e3e3',

    link: 'cornflowerblue',
    linkActive: '#09b8f3',

    success: 'lightgreen',
    primary: 'lightskyblue',
    error: '#b51d1d',
  }
}
