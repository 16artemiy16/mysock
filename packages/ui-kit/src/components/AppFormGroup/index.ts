import AppFormGroup from './AppFormGroup';
import AppFormGroupControl from './AppFormGroupControl';

AppFormGroup.Control = AppFormGroupControl;

export default AppFormGroup;
