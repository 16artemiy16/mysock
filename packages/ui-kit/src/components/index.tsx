export { default as AppBtn } from './AppBtn/AppBtn';
export { default as AppFormControl } from './AppFormControl/AppFormControl';
export * from './AppFormControl/validators';
export { default as AppIcon } from './AppIcon/AppIcon';
export { default as AppInput } from './AppInput/AppInput';
export { default as AppFormGroup } from './AppFormGroup';
