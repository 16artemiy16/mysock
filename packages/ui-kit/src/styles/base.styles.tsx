import { css } from 'styled-components';

export const borders = css`
  border-radius: 15px 225px 255px 15px 15px 255px 225px 15px;
  border-style: solid;
  border-width: 2px;
  box-sizing: border-box;
  border-bottom-left-radius: 15px 255px;
  border-bottom-right-radius: 225px 15px;
  border-top-left-radius: 255px 15px;
  border-top-right-radius: 15px 225px;
  border-color: ${props => props.theme.colors.black};
`;

export const base = css`
  ${borders}
  align-self: center;
  color:  ${props => props.theme.colors.black};
  cursor: pointer;
  display: inline-block;
  font-family: Neucha, sans-serif;
  font-size: 1rem;
  line-height: 23px;
  outline: none;
  padding: .75rem;
  text-decoration: none;
  transition: all 235ms ease-in-out;
  
  user-select: none;
  touch-action: manipulation;

  &:focus {
    box-shadow: rgba(0, 0, 0, .3) 2px 8px 4px -6px;
  }
`;

export const onHover = css`
  &:hover {
    transition: transform 235ms ease-in-out;
    transform: translate3d(0, 2px, 0);
  }
`;
