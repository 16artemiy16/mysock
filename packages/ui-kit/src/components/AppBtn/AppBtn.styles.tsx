import styled, { css } from 'styled-components';
import { ThemeI } from '@mysock-front/application/src/styles/theme.styles';
import { base, onHover } from '../../styles/base.styles';

interface Props {
  isLink?: boolean;
  isActive?: boolean
  theme: ThemeI;
  width?: string;
  disabled?: boolean;
}

export const Button = styled.button<Props>`
  ${base};
  ${onHover}
  width: ${props => props.width || null};
  background-color: ${props => getBackground(props)};
  cursor: ${props => props.disabled && 'not-allowed'};
  
  a {
    text-decoration: none;
    color: black;
  }
`;

function getBackground ({ isLink, isActive, theme, disabled }: Record<string, any>): string {
  const { colors } = theme;
  if (disabled) {
    return colors.grayLight;
  }
  if (!isLink) {
    return colors.white;
  }
  return isActive ? colors.linkActive : colors.link;
}
