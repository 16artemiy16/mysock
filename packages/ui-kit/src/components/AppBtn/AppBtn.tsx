import { ButtonHTMLAttributes, FC } from 'react';
import { Button } from './AppBtn.styles';
import { useMatch, useNavigate } from 'react-router-dom';

export interface AppBtnProps extends ButtonHTMLAttributes<any> {
  /** Text content of the button. **/
  children?: string;
  /** It it's a link. **/
  to?: string;
  width?: string;
}

const AppBtn: FC<AppBtnProps> = ({ children, to, width, ...rest }) => {
  const isLink = !!to;
  const isActive = !!useMatch(to || '');
  const navigate = useNavigate();

  const handleClick = (e: any) => {
    if (rest.disabled) {
      return;
    }
    const additionalAction = isLink ? () => navigate(to) : () => null;
    if (rest.onClick) {
      rest!.onClick(e)
    }
    additionalAction();
  }

  return <Button
    {...rest}
    isActive={isActive}
    isLink={isLink}
    onClick={handleClick}
    width={width}
  >
    {children}
  </Button>
}

export default AppBtn;
