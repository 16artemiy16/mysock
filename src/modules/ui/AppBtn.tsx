import { FC } from 'react';
import { Button } from './AppBtn.styles';
import { useMatch, useNavigate } from 'react-router-dom';

interface Props {
  /** Text content of the button. **/
  children?: string;
  /** It it's a link. **/
  to?: string;
}

const AppBtn: FC<Props> = ({ children, to }) => {
  const isLink = !!to;
  const isActive = !!useMatch(to as string);
  const navigate = useNavigate();

  const handleClick = isLink ? () => navigate(to) : () => null;

  return <Button isActive={isActive} isLink={isLink} onClick={handleClick}>{children}</Button>
}

export default AppBtn;
