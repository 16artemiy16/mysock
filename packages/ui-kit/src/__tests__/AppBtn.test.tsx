import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom'
import 'jest-styled-components'
import "@testing-library/jest-dom/extend-expect";

import { ThemeProvider } from 'styled-components';

import { AppBtn } from '../components';
import { theme } from '@mysock-front/application/src/styles/theme.styles';
import { AppBtnProps } from '../components/AppBtn/AppBtn';

const renderAppBtn = (props: AppBtnProps = {}, child?: string) => render(
  <ThemeProvider theme={theme}>
    <MemoryRouter>
      <AppBtn {...props}>{child}</AppBtn>
    </MemoryRouter>
  </ThemeProvider>)

describe('AppBtn', () => {
  it('should render without props', () => {
    const { container } = renderAppBtn({}, 'Click');
    expect(container).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveTextContent('Click')
  });

  describe('disabled', () => {
    const renderAppBtnDisabled = () => renderAppBtn({ disabled: true });

    it('should render with correct styles', () => {
      const { container } = renderAppBtnDisabled();

      expect(container.firstChild).toMatchSnapshot();
      expectStyles(container.firstChild, { cursor: 'not-allowed', 'background-color': theme.colors.grayLight })
    });
  });

  describe('link', () => {
    const renderAppBtnLink = (props: AppBtnProps = {}) => renderAppBtn({ to: '/test', ...props });

    it('should render with correct styles', () => {
      const {container} = renderAppBtnLink();

      expect(container.firstChild).toMatchSnapshot();
      expectStyles(container.firstChild, { 'background-color': theme.colors.link })
    });
  })
});
