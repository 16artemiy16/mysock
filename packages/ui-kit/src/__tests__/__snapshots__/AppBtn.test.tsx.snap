// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBtn disabled should render with correct styles 1`] = `
.c0 {
  border-radius: 15px 225px 255px 15px 15px 255px 225px 15px;
  border-style: solid;
  border-width: 2px;
  box-sizing: border-box;
  border-bottom-left-radius: 15px 255px;
  border-bottom-right-radius: 225px 15px;
  border-top-left-radius: 255px 15px;
  border-top-right-radius: 15px 225px;
  border-color: black;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: black;
  cursor: pointer;
  display: inline-block;
  font-family: Neucha,sans-serif;
  font-size: 1rem;
  line-height: 23px;
  outline: none;
  padding: .75rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 235ms ease-in-out;
  transition: all 235ms ease-in-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  touch-action: manipulation;
  background-color: #e6e3e3;
  cursor: not-allowed;
}

.c0:focus {
  box-shadow: rgba(0,0,0,.3) 2px 8px 4px -6px;
}

.c0:hover {
  -webkit-transition: -webkit-transform 235ms ease-in-out;
  -webkit-transition: transform 235ms ease-in-out;
  transition: transform 235ms ease-in-out;
  -webkit-transform: translate3d(0,2px,0);
  -ms-transform: translate3d(0,2px,0);
  transform: translate3d(0,2px,0);
}

.c0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: black;
}

<button
  class="c0"
  disabled=""
/>
`;

exports[`AppBtn link should render with correct styles 1`] = `
.c0 {
  border-radius: 15px 225px 255px 15px 15px 255px 225px 15px;
  border-style: solid;
  border-width: 2px;
  box-sizing: border-box;
  border-bottom-left-radius: 15px 255px;
  border-bottom-right-radius: 225px 15px;
  border-top-left-radius: 255px 15px;
  border-top-right-radius: 15px 225px;
  border-color: black;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: black;
  cursor: pointer;
  display: inline-block;
  font-family: Neucha,sans-serif;
  font-size: 1rem;
  line-height: 23px;
  outline: none;
  padding: .75rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 235ms ease-in-out;
  transition: all 235ms ease-in-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  touch-action: manipulation;
  background-color: cornflowerblue;
}

.c0:focus {
  box-shadow: rgba(0,0,0,.3) 2px 8px 4px -6px;
}

.c0:hover {
  -webkit-transition: -webkit-transform 235ms ease-in-out;
  -webkit-transition: transform 235ms ease-in-out;
  transition: transform 235ms ease-in-out;
  -webkit-transform: translate3d(0,2px,0);
  -ms-transform: translate3d(0,2px,0);
  transform: translate3d(0,2px,0);
}

.c0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: black;
}

<button
  class="c0"
/>
`;

exports[`AppBtn should render without props 1`] = `
.c0 {
  border-radius: 15px 225px 255px 15px 15px 255px 225px 15px;
  border-style: solid;
  border-width: 2px;
  box-sizing: border-box;
  border-bottom-left-radius: 15px 255px;
  border-bottom-right-radius: 225px 15px;
  border-top-left-radius: 255px 15px;
  border-top-right-radius: 15px 225px;
  border-color: black;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: black;
  cursor: pointer;
  display: inline-block;
  font-family: Neucha,sans-serif;
  font-size: 1rem;
  line-height: 23px;
  outline: none;
  padding: .75rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 235ms ease-in-out;
  transition: all 235ms ease-in-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  touch-action: manipulation;
  background-color: white;
}

.c0:focus {
  box-shadow: rgba(0,0,0,.3) 2px 8px 4px -6px;
}

.c0:hover {
  -webkit-transition: -webkit-transform 235ms ease-in-out;
  -webkit-transition: transform 235ms ease-in-out;
  transition: transform 235ms ease-in-out;
  -webkit-transform: translate3d(0,2px,0);
  -ms-transform: translate3d(0,2px,0);
  transform: translate3d(0,2px,0);
}

.c0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: black;
}

<button
  class="c0"
>
  Click
</button>
`;
